### PR TITLE
Add UnifyFS Citation sections to docs and readme

### PR DESCRIPTION
**Pull request by CamStan**
Also discovered we can add a "Cite this repository" button to the repo sometime if desired. It can automatically provide different formats for the citation.

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,5 +29,10 @@ Status of UnifyFS development branch (dev):
 
 [![Read the Docs](https://readthedocs.org/projects/unifyfs/badge/?version=dev)](https://unifyfs.readthedocs.io)
 
+## UnifyFS Citation
+We recommend that you use this citation for UnifyFS:
+
+  * Michael Brim, Adam Moody, Seung-Hwan Lim, Ross Miller, Swen Boehm, Cameron Stanavige, Kathryn Mohror, Sarp Oral, “UnifyFS: A User-level Shared File System for Unified Access to Distributed Local Storage,” International Parallel & Distributed Processing Symposium (IPDPS) 2023, St. Petersburg, FL, May 2023.
```
Comment (MichaelBrim) [inline README.md:35]: Put "37th IEEE" before "International Parallel ..."

File: docs/overview.rst [hunk 2]
```diff
@@ -52,3 +53,14 @@ the job allocation. The UnifyFS server handles the I/O
...
+-----
+
+----------------
+UnifyFS Citation
+----------------
+
+We recommend that you use this as the primary citation for UnifyFS as well as a
+reference for further details on the UnifyFS architecture and semantics:
+
+    Michael Brim, Adam Moody, Seung-Hwan Lim, Ross Miller, Swen Boehm, Cameron Stanavige, Kathryn Mohror, Sarp Oral, “UnifyFS: A User-level Shared File System for Unified Access to Distributed Local Storage,” International Parallel & Distributed Processing Symposium (IPDPS) 2023, St. Petersburg, FL, May 2023.
```
Comment (MichaelBrim) [inline docs/overview.rst:66]: same comment as above

**Review by MichaelBrim — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (CamStan) [reply README.md]: Added

File: docs/overview.rst (hunk 2 above)
Comment (CamStan) [reply docs/overview.rst]: Added

File: README.md [hunk 3]
```diff
@@ -29,5 +29,10 @@ Status of UnifyFS development branch (dev):
 
 [![Read the Docs](https://readthedocs.org/projects/unifyfs/badge/?version=dev)](https://unifyfs.readthedocs.io)
 
+## UnifyFS Citation
+We recommend that you use this citation for UnifyFS:
+
+  * Michael Brim, Adam Moody, Seung-Hwan Lim, Ross Miller, Swen Boehm, Cameron Stanavige, Kathryn Mohror, Sarp Oral, “UnifyFS: A User-level Shared File System for Unified Access to Distributed Local Storage,” 37th IEEE International Parallel & Distributed Processing Symposium (IPDPS) 2023, St. Petersburg, FL, May 2023.
```
Comment (MichaelBrim) [inline README.md:35]: One more thing. It should be "(IPDPS 2023)", rather than "(IPDPS) 2023". Same change in both places again.

Comment (CamStan) [reply README.md]: Updated

**Review by kathrynmohror — APPROVED** [review]

**Review by MichaelBrim — APPROVED** [review]

**Pull request merged**